src/components/UserSelector.tsx

```diff
@@ -1,16 +1,76 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+type Props = {
+  selectedUser: User | null;
+  onSelect: (v: User | null) => void;
+  onPost: (v: null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelect,
+  onPost,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [menuActive, setMenuActive] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  useEffect(() => {
+    function handleClickOutside(e: MouseEvent) {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(e.target as Node)
+      ) {
+        setMenuActive(false);
+      }
+    }
+
+    document.addEventListener('click', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, []);
+
+  const handleUserChange = (
+    e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+  ) => {
+    const currentUser = e.target as HTMLAnchorElement;
```

Comment:
The `handleUserChange` function assumes that the event target is always an anchor element. Consider adding a check to ensure `currentUser` is an anchor element before accessing its attributes to prevent potential runtime errors.